datasets/dataset_loading_hub.py

```diff
@@ -6,35 +6,46 @@
 
 from torch.utils.data import Subset
 
-from datasets.custom_transforms import SimCLRTransform
+from datasets.custom_transforms import *
 from utils.cinic_utils import enlarge_cinic_10, download_cinic
 from rich import print
 
 ImageShape = namedtuple("ImageShape", ["channels", "width", "height"])
 
 
 class MNISTLoader:
-    def __init__(self):
+    def get_default_transforms(self):
```

Comment:
Can you change this to be `__init__` (or change all the others to `get_default_transforms` for consistency?

My feeling is that we should prefer to have these instantiated in the `__init__` function and not return anything, because by default, the class should be entirely responsible for the pre-setup data transforms. Thoughts?

Reply:
^ ah sorry hadn't looked through. 

Can you give a reason why you think `get_default_transforms` should act like this instead of the `__init__` behaviour we had before?

Reply:
(although I think you have missed out ImageNet)

Reply:
In some experiments where I needed to add my own transforms, having a function to build this allows me flexibility to do things I couldn't do before. For example, I could choose if I wanted my new transforms before or after the defaults among other things.

Reply:
Will do this to ImageNet too. (I transferred this feature from another repo where this became necessary to do, kinda forgot ImageNet somehow, sorry!)

Reply:
In the previous set up the workflow was all done in the `load_datasets` function:
1. call `__init__`, which generates the default transforms 
2. [optional] make any modifications (e.g. replace transforms with SimCLR transforms, prepend new transforms, append new transforms) 
3. call `get_data` to get the data with the [optionally modified] transforms 

Now it seems more complex: `get_data` automatically calls `get_transforms`, so to add a transform I have to modify the `get_data` function? But then what if I want that change to apply to all of the datasets? I have to modify all of the `get_data`s? or is there a neater way I am missing?

Reply:
Give me a moment. I was considering something more streamlined. I'll commit a new proposal soon.

Reply:
Here's an idea. It makes things a lot easier to manipulate if you have a class that gets the transforms, and another that can set the transforms (or change them). 

Also, we should be replicating transform settings as much as possible. For example, certain transforms are repeated in CIFAR10/100. Only diff are normalization constants. Should we have a CIFARtransform prototype that takes as arguments the mean and stds? This way we can reuse those transforms without copy pasting so many lines.